core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -708,30 +711,35 @@ object ConsumerGroupCommand extends Logging {
       options.timeoutMs(t)
     }
 
-    private def parseTopicPartitionsToReset(groupId: String, topicArgs: Seq[String]): Seq[TopicPartition] = topicArgs.flatMap {
-      case topicArg if topicArg.contains(":") =>
-        val topicPartitions = topicArg.split(":")
+    private def parseTopicPartitionsToReset(topicArgs: Seq[String]): Seq[TopicPartition] = {
+      val (topicsWithPartitions, topics) = topicArgs.partition(_.contains(":"))
+      val specifiedPartitions = topicsWithPartitions.flatMap { arg =>
+        val topicPartitions = arg.split(":")
```

Comment:
How about using match pattern? 
```scala
        arg.split(":") match {
          case Array(topic, partitions) =>
            partitions.split(",").map(partition => new TopicPartition(topic, partition.toInt))
        }
```
The benefit is that it can throw exception if length of `split(":")` is not equal to 2.

Reply:
This is not great on its own since the error will be cryptic. It's ok to use the pattern match, but you would want to handle the case where it doesn't match with an appropriate error message.

Reply:
I see that Rajini has the extra case in her changes, so all good.